test/services/objection.session.service.spec.unit.ts

```diff
@@ -1,10 +1,37 @@
 import { Session } from "ch-node-session-handler";
-import { retrieveAccessTokenFromSession } from "../../src/services/objection.session.service";
+import { IUserProfile } from "ch-node-session-handler/lib/session/model/SessionInterfaces";
+import {
+  retrieveAccessTokenFromSession,
+  retrieveUserProfileFromSession,
+} from "../../src/services/objection.session.service";
 
 const accessTokenValue = "tokenABC123";
 
 describe ("objections session service tests", () => {
 
+  it("should retrieve user profile when present in the session", () => {
+    const session: Session = new Session();
+    session.data = {
+      signin_info: {
+        user_profile: {
+          email: "demo@ch.gov.uk",
+          forename: "Joe",
+          surname: "Bloggs",
+        },
+      },
+    };
+    const userProfile: IUserProfile = retrieveUserProfileFromSession(session);
+    expect(userProfile).not.toBeUndefined();
+    expect(userProfile).toEqual({email: "demo@ch.gov.uk", forename: "Joe", surname: "Bloggs"});
+  });
+
+  it("should throw error when user profile is absent", () => {
+    const session: Session = new Session();
+    expect(() => {
```

Comment:
this throws when there is no signIninfo, to test if it throws when there is no user profile you would need a session with signInInfo but no user profile.

Reply:
Updated tests.